### PR TITLE
Added PlayerLog

### PR DESCRIPTION
**Pull request by dheld**

### CODE REVIEW

**Comment by Tazshelby2016** [comment]
Could I get a description of what is done here for the sake of non-programmers?

File: core/src/io/anuke/mindustry/world/blocks/storage/Unloader.java [hunk 1]
```diff
@@ -50,6 +52,7 @@ public void playerPlaced(Tile tile){
     public void configured(Tile tile, Player player, int value){
         tile.entity.items.clear();
         tile.<UnloaderEntity>entity().sortItem = content.item(value);
+        playerLog.record(player, PlayerLog.Action.configured, tile, null);
```
Comment (Anuken) [inline core/src/io/anuke/mindustry/world/blocks/storage/Unloader.java]: This should be recorded after each configure at the end of `InputHandler#onTileConfig`, not in each block type.

File: core/src/io/anuke/mindustry/game/PlayerLog.java [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+        public static String list(){
+            final StringBuilder sb = new StringBuilder();
+            for(Action value : values()){
+                sb.append(", ").append(value);
+            }
+            return sb.toString().substring(2);
+        }
+    }
+    /** The size of the log; for 16 players averaging 200 APM this will save about 5 min of actions */
+    public static final int BUFFER_SIZE = 16*1024;
```
Comment (Anuken) [inline core/src/io/anuke/mindustry/game/PlayerLog.java]: This should be in `camelCase`. I think I might have mentioned it earlier. Yes, it's not standard capitalization, but I believe making harmless constant all caps is ugly, and it doesn't match the rest of the codebase.

File: core/src/io/anuke/mindustry/game/PlayerLog.java [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+    /** Events per "page" when returning results */
+    public static final int PAGE_SIZE = 16;
```
Comment (Anuken) [inline core/src/io/anuke/mindustry/game/PlayerLog.java]: `camelCase`

File: core/src/io/anuke/mindustry/game/PlayerLog.java [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+ */
+public class PlayerLog{
+    public enum Action{
+        constructed,
+        deconstructed,
+        linked,
+        unlinked,
+        configured;
+
+        public static final String LIST = list();
```
Comment (Anuken) [inline core/src/io/anuke/mindustry/game/PlayerLog.java]: `camelCase`

File: core/src/io/anuke/mindustry/world/blocks/BuildBlock.java [hunk 5]
```diff
@@ -51,8 +51,10 @@ public static void onDeconstructFinish(Tile tile, Block block, int builderID){
         Team team = tile.getTeam();
         Effects.effect(Fx.breakBlock, tile.drawx(), tile.drawy(), block.size);
         world.removeBlock(tile);
-        Events.fire(new BlockBuildEndEvent(tile, playerGroup.getByID(builderID), team, true));
+        final Player player = playerGroup.getByID(builderID);
```
Comment (Anuken) [inline core/src/io/anuke/mindustry/world/blocks/BuildBlock.java]: Any particular reason you're making this player final? It doesn't make the code run faster, and syntax highlighting should let you know if you accidentally mutate it somewhere anyway.

File: tests/src/test/java/io/anuke/mindustry/game/PlayerLogTest.java [hunk 6]
```diff
@@ -0,0 +1,231 @@
...
+    private PlayerEvent createEvent(short xy){
+        return new PlayerEvent(TIMESTAMP, PLAYER_NAME, PlayerLog.Action.linked, TILE1_NAME, xy, xy);
+    }
+
+    private PlayerEvent createEvent(short xy1, short xy2){
+        return new PlayerEvent(TIMESTAMP, PLAYER_NAME, PlayerLog.Action.linked, TILE1_NAME, xy1, xy1, TILE2_NAME, xy2, xy2);
+    }
+
+    @Test
+    public void append_Single_Succeeds(){
```
Comment (Anuken) [inline tests/src/test/java/io/anuke/mindustry/game/PlayerLogTest.java]: Why use underscores here?

**Comment by Anuken** [comment]
Closing due to stale request; will be implement differently.

**Pull request closed without merging**